src/libraries/System.Collections/tests/Generic/Stack/Stack.Generic.Tests.cs

```diff
@@ -314,11 +314,14 @@ public void Stack_Generic_EnsureCapacity_NegativeCapacityRequested_Throws()
             AssertExtensions.Throws<ArgumentOutOfRangeException>("capacity", () => stack.EnsureCapacity(-1));
         }
 
-        const int MaxArraySize = 0X7FEFFFFF;
+        public static IEnumerable<object[]> Stack_Generic_EnsureCapacity_LargeCapacityRequested_Throws_MemberData()
+        {
+            yield return new object[] { Array.MaxLength + 1 };
+            yield return new object[] { int.MaxValue };
+        }
 
         [Theory]
-        [InlineData(MaxArraySize + 1)]
-        [InlineData(int.MaxValue)]
+        [MemberData(nameof(Stack_Generic_EnsureCapacity_LargeCapacityRequested_Throws_MemberData))]
         [SkipOnMono("mono forces no restrictions on array size.")]
```

Comment:
Should we fix this for Mono, or at least have a bug opened to fix this for Mono?

Reply:
I opened https://github.com/dotnet/runtime/issues/51411